### PR TITLE
Fix helper functions for MySQL syntax

### LINKED ISSUE #3870: Migration fails in latest version.
### Component

server

### Describe the bug

A migration fails in the latest version of Woodpecker.

The server wouldn't startup and crashes while attempting to finish the migration.

Last working version is `next-2fa9432ef8`

### Steps to reproduce

1. Set up Woodpecker server with MariaDB server as database.
2. Run docker compose and see logs.

### Expected behavior

_No response_

### System Info

```shell
Woodpecker version: next-e118f8d980
MariaDB 10.11.6
```

### Additional context

```
{"level":"info","time":"2024-07-03T21:30:09Z","message":"log level: info"}
{"level":"fatal","error":"can't setup store: could not migrate datastore: migration alter-table-registries-fix-required-fields failed: Error 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'COLUMN `repo_id` SET NOT NULL' at line 1","time":"2024-07-03T21:30:09Z","message":"error running server"}
```

### Validations

- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]

### PR DESCRIPTION
**Pull request by lafriks**
Fixes #3870

Would be nice if someone could test it as I have no mysql/mariadb instances to reproduce this on

### CODE REVIEW

**Comment by rubenelshof** [comment]
I can help testing it if you want.
Please let me know if I have to do something specifically for the test.

**Comment by 6543** [comment]
> I can help testing it if you want. Please let me know if I have to do something specifically for the test.

the pipeline should now publish an image you can test against

**Comment by rubenelshof** [comment]
I just tested it and it fails with the following error.
```
8:58AM INF woodpecker/src/github.com/woodpecker-ci/woodpecker/shared/logger/logger.go:101 > log level: debug
8:58AM FTL woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/server/main.go:46 > error running server error="can't setup store: could not migrate datastore: migration alter-table-registries-fix-required-fields failed: Error 1054 (42S22): Unknown column 'column_name' in 'where clause'"
```

**Comment by lafriks** [comment]
as soon as CI builds new image you can try again, should be fixed now

**Comment by rubenelshof** [comment]
It failed again.
```
10:14AM INF woodpecker/src/github.com/woodpecker-ci/woodpecker/shared/logger/logger.go:101 > log level: debug
10:14AM FTL woodpecker/src/github.com/woodpecker-ci/woodpecker/cmd/server/main.go:46 > error running server error="can't setup store: could not migrate datastore: migration alter-table-registries-fix-required-fields failed: column repo_id data type in table registries can not be detected"
```

**Comment by lafriks** [comment]
could be resultset is actually not lowercase as I thought, will try to use `Type` instead of `type`

**Comment by lafriks** [comment]
please test again when new version is built

**Comment by rubenelshof** [comment]
This works and fixes the issue.
No errors reported in the logs.

**Comment by lafriks** [comment]
can you show output of `show columns from registries;` after migrations?

**Comment by rubenelshof** [comment]
> can you show output of `show columns from registries;` after migrations?

```
+----------+---------------+------+-----+---------+----------------+
| Field    | Type          | Null | Key | Default | Extra          |
+----------+---------------+------+-----+---------+----------------+
| id       | bigint(20)    | NO   | PRI | NULL    | auto_increment |
| repo_id  | bigint(20)    | NO   | MUL | 0       |                |
| address  | varchar(255)  | NO   | MUL | NULL    |                |
| username | varchar(2000) | YES  |     | NULL    |                |
| password | text          | YES  |     | NULL    |                |
| org_id   | bigint(20)    | NO   | MUL | 0       |                |
+----------+---------------+------+-----+---------+----------------+
```

**Comment by lafriks** [comment]
Great! Everything looks as it should be. Thanks for your help on testing it! ❤️

File: server/store/datastore/migration/common.go [hunk 1]
```diff
@@ -185,7 +185,26 @@ func dropTableColumns(sess *xorm.Session, tableName string, columnNames ...strin
...
+		if err != nil {
+			return err
+		}
+
+		if len(res) == 0 || len(res[0]["Type"]) == 0 {
+			return fmt.Errorf("column %s data type in table %s can not be detected", column, table)
+		}
+
+		dataType := string(res[0]["Type"])
+		var nullable string
```
[the remark this answers is not in the document]
Comment (6543) [reply server/store/datastore/migration/common.go:200]: ah yes we have to carry that info onwards ...

**Review by 6543 — APPROVED** [review]

**Pull request merged**